### PR TITLE
Many Markdown tweaks

### PR DESCRIPTION
**Pull request by XhmikosR**
I know this PR is quite big, but I went one by one file and applied many Markdown tweaks.

1. #195 should be merged beforehand
2. I tried to specify the code blocks language in many cases; there are some cases I'm not 100% sure what to use. For example for pi-hole.conf
3. when the codeblock is supposed to be raw text I used `text`; not specifying a codebclock language results in wrong highlighting if the highlighting library cannot autodetect the language. So we need to make sure we add it all the time. **EDI**: I guess we could set `guess_lang: false` for `codehilite`

Finally, in order for this to be foolproof we need some kind of markdown linting added later. I couldn't find a maintained Python lib, and personally I use https://github.com/DavidAnson/markdownlint, but there are probably other solutions out there

Live preview: <https://optimistic-ptolemy-ed7ff3.netlify.com/>

### CODE REVIEW

File: docs/core/pihole-command.md [hunk 1]
```diff
@@ -57,12 +61,13 @@ See [Regex Blocking](/ftldns/regex/overview/) for more information about using R
 * Each domain is validated using regex (except when using `-regex`), to ensure invalid domains and IDNs are not added
 * A whitelisted domain gets added or removed from `/etc/pihole/whitelist.txt`
 * A blacklisted domain gets added or removed from `/etc/pihole/blacklist.txt`
-  * On either list type, `gravity.sh` is then called to consolidate an updated copy of `gravity.list`, and the DNS server is reloaded
+    * On either list type, `gravity.sh` is then called to consolidate an updated copy of `gravity.list`, and the DNS server is reloaded
```
Comment (XhmikosR) [inline docs/core/pihole-command.md]: I assume this was meant to make it a nested list

**Comment by XhmikosR** [comment]
I have a WIP branch which adds Markdownlint https://github.com/pi-hole/docs/compare/master...XhmikosR:markdownlint. I'll proceed with this later.

**Comment by XhmikosR** [comment]
I'll freeze the changes here until we land this. The markdownlint PR will take care of most of the other issues, except for the highlight autoguess issue, which IMHO we should disable and be explicit.

A few comparisons:

* https://docs.pi-hole.net/guides/nginx-configuration/
* https://optimistic-ptolemy-ed7ff3.netlify.com/guides/nginx-configuration/

* https://docs.pi-hole.net/guides/benchmark/
* https://optimistic-ptolemy-ed7ff3.netlify.com/guides/benchmark/

* https://docs.pi-hole.net/ftldns/compile/
* https://optimistic-ptolemy-ed7ff3.netlify.com/ftldns/compile/

* https://docs.pi-hole.net/guides/vpn/clients/
* https://optimistic-ptolemy-ed7ff3.netlify.com/guides/vpn/clients/

File: docs/guides/tor/overview.md [hunk 2]
```diff
@@ -2,34 +2,37 @@
...
 
-   So in the end it boils down to one of the following use cases.
-   
+So in the end it boils down to one of the following use cases.
 
-   - Encrypt your DNS traffic using Tor so your ISP can't collect it (but still is able to [collect what Websites/IPs you visit](#%EF%B8%8F-attention-%EF%B8%8F) unless you route that traffic also over Tor) and the DNS Server won't see your real IP for the price of maybe getting a Bad Exit Node that fakes answers to DNS queries.
+- Encrypt your DNS traffic using Tor so your ISP can't collect it (but still is able to [collect what Websites/IPs you visit](#%EF%B8%8F-attention-%EF%B8%8F) unless you route that traffic also over Tor) and the DNS Server won't see your real IP for the price of maybe getting a Bad Exit Node that fakes answers to DNS queries.
```
Comment (XhmikosR) [inline docs/guides/tor/overview.md]: I just noticed this weird anchor. Does anybody know where it was supposed to point to? `#%EF%B8%8F-attention-%EF%B8%8F` Not sure if there are more cases

Comment (DL6ER) [reply docs/guides/tor/overview.md]: We can try to find this out later by checking `git blame`. There might be useful information hidden here.

Comment (XhmikosR) [reply docs/guides/tor/overview.md]: I did and it seems it was just like that when migrated hence https://github.com/pi-hole/docs/pull/199

**Pull request merged**